auth_totp/wizards/res_users_authenticator_create.py

```diff
@@ -0,0 +1,115 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 LasLabs Inc.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import logging
+import urllib
+from openerp import api, fields, models
+from openerp.exceptions import ValidationError
+
+_logger = logging.getLogger(__name__)
+try:
+    import pyotp
+except ImportError:
+    _logger.debug(
+        'Could not import PyOTP. Please make sure this library is available in'
+        ' your environment.'
+    )
+
+
+class ResUsersAuthenticatorCreate(models.TransientModel):
+    _name = 'res.users.authenticator.create'
+    _description = 'MFA App/Device Creation Wizard'
+
+    name = fields.Char(
+        string='Authentication App/Device Name',
+        help='A name that will help you remember this authentication'
+        ' app/device',
+        required=True,
+        index=True,
+    )
+    secret_key = fields.Char(
+        default=lambda s: pyotp.random_base32(),
+    )
+    qr_code_tag = fields.Html(
+        compute='_compute_qr_code_tag',
+        string='QR Code',
+        help='Scan this image with your authentication app to add your account'
+    )
+    user_id = fields.Many2one(
+        comodel_name='res.users',
+        default=lambda s: s._default_user_id(),
+        required=True,
+        string='Associated User',
+        help='This is the user whose account the new authentication app/device'
+        ' will be tied to',
+        readonly=True,
+        index=True,
+    )
+    confirmation_code = fields.Char(
+        string='Confirmation Code',
+        help='Enter the latest six digit code generated by your authentication'
+        ' app',
+        required=True,
+    )
+
+    @api.model
+    def _default_user_id(self):
+        user_id = self.env.context.get('uid')
+        return self.env['res.users'].browse(user_id)
+
+    @api.multi
+    @api.depends(
+        'secret_key',
+        'user_id.display_name',
+        'user_id.company_id.display_name',
+    )
+    def _compute_qr_code_tag(self):
+        for record in self:
```

Comment:
There was one missing a test case, but I can't identify it until the tests are passing. As long as the negative case is tested, the more explicit is fine.